src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyQueueCapacityAction.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.CPU;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.NETWORK;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ModifyQueueCapacityAction implements Action {
+
+  public static final String NAME = "modify_queue_capacity";
+  public static final int COOL_OFF_PERIOD_IN_SECONDS = 300;
```

Comment:
it may be ok for the first cut, but we may have to make this param configurable for different actions

Reply:
Its not clear what we mean by configurable here -

1. Configurable across different instances of `ModifyQueueCapacityAction` ? 
I think it would be the same for a particular type of action, we would want to wait `x` amount of time when modifying queues. Possibly different wait times for increase v/s decrease queue capacity, but that logic (waiting for queue drain out) can lie within `coolOffPeriodInSeconds()`. I'm not sure why different instances of `ModifyQueueCapacityAction` will need different cool off periods.

2. Configurable across different types of action, like modifyCacheBound/modifyQueueCapacity etc. -
Each such action class will have its own cool off period defined.

3. Configurable via `rca.conf` -
We can add this logic in `coolOffPeriodInSeconds` when needed, but I'm still skeptical if we want to add another knob in `rca.conf`. It gets hard to manually maintain all these configurations. Instead, the function might return a cool off based on some calculations, like cool off for moveShard may depend on shard size and recovery speed configured, rather than a manual setting per cluster.